### ISSUE TITLE
CARLA object detection dataset has incorrect ground truth labels

### ISSUE THREAD

**Issue by Lodour** [twosixlabs/armory#1206]
I saved some images and bounding boxes from the CARLA Object Detection dataset, then I found that many ground truth boxes and labels are incorrect.

I extract data using the following two functions
```python
from armory.data.adversarial_datasets import carla_obj_det_dev
from armory.data.datasets import carla_obj_det_train
```

See below for some examples of incorrect labels.

I have added the provided bounding boxes, whose labels are indicated by different colors:
* Red: 1 Pedestrian
* Green: 2 Vehicle
* Blue: 3 Traffic lights
* White: 4 Green screen

**dev split, index 20, the vehicle is labeled as traffic lights (blue).**
![dev-020-rgb](https://user-images.githubusercontent.com/11145354/139755730-489d9384-f858-4756-904f-b3dcedc1fd7f.png)
![dev-020-depth](https://user-images.githubusercontent.com/11145354/139756101-7cc7a187-b11f-48f3-a08e-8c15d434eb28.png)

**train split, index 100, the vehicle is labeled as pedestrain (red), many small random bounding boxes.**
![train-100-depth](https://user-images.githubusercontent.com/11145354/139756247-42aec823-4e25-4b11-a1ee-26ca538a7d7b.png)
![train-100-rgb](https://user-images.githubusercontent.com/11145354/139756248-102c0cab-9217-4c4d-a77f-30484c863d70.png)

**Comment by yusong-tan** [comment]
@Lodour this is a shortcoming that we are aware of in CARLA. When multiple objects overlap (i.e., connected pixel-wise), CARLA will treat them all together as one class and produce one semantic segmentation. We currently cannot separate out the individual objects that overlap. Relatively, this doesn't happen very often in our training data and shouldn't significantly impact training.

In your examples, either the cars overlap with background traffic lights, or cars overlap with pedestrians. In both cases, CARLA will pick one class and label all the overlapping objects as the same class.

**Comment by Lodour** [comment]
Thanks for the clarification.
